core/src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -39,14 +38,12 @@ trait CassandraRecovery extends CassandraTagRecovery with TaggedPreparedStatemen
       replayCallback: PersistentRepr => Unit): Future[Unit] = {
     log.debug("asyncReplayMessages pid {} from {} to {}", persistenceId, fromSequenceNr, toSequenceNr)
 
-    val persistentActor = sender()
```

Comment:
this is the bug, because AsyncWriteJournal is calling asyncReplayMessages from Future flatMap

Reply:
:+1: